dexbot/strategies/relative_orders.py

```diff
@@ -118,57 +176,108 @@ def update_orders(self):
+        need_update = False
         if not orders:
-            self.update_orders()
+            need_update = True
         else:
-            orders_changed = False
-
             # Loop trough the orders and look for changes
             for order_id, order in orders.items():
                 current_order = self.get_order(order_id)
 
                 if not current_order:
-                    orders_changed = True
-                    self.write_order_log(self.worker_name, order)
+                    need_update = True
+                    self.log.debug('Could not found order on the market, it was filled, expired or cancelled')
+                    # Write a trade log entry only when we are not using custom expiration because we cannot
+                    # distinguish an expired order from filled
+                    if not self.is_custom_expiration:
+                        self.write_order_log(self.worker_name, order)
+                elif self.is_reset_on_partial_fill:
+                    # Detect partially filled orders;
+                    # on fresh order 'for_sale' is always equal to ['base']['amount']
+                    if current_order['for_sale']['amount'] != current_order['base']['amount']:
+                        diff_abs = current_order['base']['amount'] - current_order['for_sale']['amount']
+                        diff_rel = diff_abs / current_order['base']['amount']
+                        if diff_rel >= self.partial_fill_threshold:
+                            need_update = True
+                            self.log.info('Partially filled order detected, filled {:.2%}'.format(diff_rel))
+                            # FIXME: need to write trade operation; possible race condition may occur: while
+                            #        we're updating order it may be filled futher so trade log entry will not
+                            #        be correct
+
+        if (self.is_reset_on_market_trade and
+            isinstance(event, FilledOrder)):
+            self.log.debug('Market trade detected, updating orders')
+            need_update = True
+
+        if self.is_reset_on_price_change:
+            center_price = self.calculate_center_price(
+                None,
+                self.is_asset_offset,
+                self.spread,
+                self['order_ids'],
+                self.manual_offset
+            )
+            diff = (self.center_price - center_price) / self.center_price
+            diff = abs(diff)
+            if diff >= self.price_change_threshold:
+                self.log.debug('Center price changed, updating orders. Diff: {:.2%}'.format(diff))
+                need_update = True
 
-            if orders_changed:
-                self.update_orders()
-            else:
-                self.log.info("Orders correct on market")
+        if need_update:
+            self.update_orders()
+        elif self.initializing:
```

Comment:
Actually you could just check for event == 'init', since you're passing that in the constructor.

Reply:
This will not work because 'init' event is passed during initialization only, and thus code under `if need_update` will be used. On the next check en event will be a real market event, not 'init'.

Reply:
True, didn't think of that

Reply:
Actually, the point of the log message is to display it only during initialization, it doesn't matter if it doesn't get ran after that

Reply:
During initialization there will be another message "Done placing orders" whether everything fine.

Reply:
Well not really, if in the initialization phase the worker doesn't need to place any orders so it doesn't print the message. But it's fine since already removed the event parameter

Reply:
So now further changes required?

Reply:
Yeah, merged the changes already